range.c

```diff
@@ -2149,6 +2149,30 @@ range_count(int argc, VALUE *argv, VALUE range)
     }
 }
 
+/*
+ *  call-seq:
+ *    overlap?(range) -> true or false
+ *
+ *  Returns +true+ if +range+ overlaps with +self+, +false+ otherwise:
+ *
+ *    (0..2).overlap?(1..3) #=> true
+ *    (0..2).overlap?(3..4) #=> false
+ *    (0..).overlap?(..0)   #=> true
+ *
+ *  Related: Range#cover?.
+ */
+
+static VALUE
+range_overlap(VALUE range, VALUE other)
+{
+    VALUE beg_self, beg_other;
+
+    beg_self = RANGE_BEG(range);
+    beg_other = RANGE_BEG(other);
```

Comment:
`(0..2).overlap?(nil)` segfaults.

Reply:
Thank you. Fxi to raise TypeError when a non-range argument is passed.